### PR TITLE
community: add Kinetica LLM wrapper

### PR DESCRIPTION
**Pull request by chadj2**
**Description:** Initial pull request for Kinetica LLM wrapper
**Issue:** N/A
**Dependencies:** No new dependencies for unit tests. Integration tests require gpudb, typeguard, and faker
**Twitter handle:** @chad_juliano

Note: There is another pull request for Kinetica vectorstore. Ultimately we would like to make a partner package but we are starting with a community contribution.

### CODE REVIEW

File: libs/community/langchain_community/chat_models/kinetica.py [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+
+    When you create a chain from the chat prompt and execute it, the Kinetica LLM will
+    generate SQL from the input. Optionally you can use ``KineticaSqlOutputParser`` to
+    execute the SQL and return the result as a dataframe.
+
+    The following example creates an LLM using the environment variables for the
+    Kinetica connection. This will fail if the API is unable to connect to the database.
+
+    Example:
+        .. code-block:: python
```
Comment (baskaryan) [inline libs/community/langchain_community/chat_models/kinetica.py:324]: need new line after this for example to render correctly in api reference docs

File: libs/community/langchain_community/chat_models/kinetica.py [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+        result = os.getenv(name)
+        if result is not None:
+            return result
+
+        raise ValueError(
+            f"Parameter was not passed and not found in the environment: {name}"
+        )
+
+
+class KineticaChatLLM(BaseChatModel):
```
[the remark this answers is not in the document]
Comment (chadj2) [reply libs/community/langchain_community/chat_models/kinetica.py]: Class is renamed.

File: libs/community/langchain_community/chat_models/kinetica.py (hunk 1 above)
Comment (chadj2) [reply libs/community/langchain_community/chat_models/kinetica.py:324]: This is fixed.

**Comment by chadj2** [comment]
I resolved some lint errors in Kinetica notebook.

**Review by baskaryan — APPROVED** [review]

**Pull request merged**